modules/Walking_module/src/WalkingQPInverseKinematics_osqp.cpp

```diff
@@ -17,463 +17,193 @@
-    // the joint angle
-    m_hessianEigenDense = Eigen::MatrixXd(iDynTree::toEigen(m_jointRegulatizationHessian)) +
-        iDynTree::toEigen(m_neckJacobian).transpose() *
-        iDynTree::toEigen(m_neckWeightMatrix) *
-        iDynTree::toEigen(m_neckJacobian);
-
-    if(!m_useCoMAsConstraint)
-    {
-        m_hessianEigenDense =  m_hessianEigenDense + iDynTree::toEigen(m_comJacobian).transpose() *
-            iDynTree::toEigen(m_comWeightMatrix) * iDynTree::toEigen(m_comJacobian);
-    }
-
-    Eigen::SparseMatrix<double> hessianEigen = m_hessianEigenDense.sparseView();
-
-    if(m_optimizerSolver->isInitialized())
-    {
-        if(!m_optimizerSolver->updateHessianMatrix(hessianEigen))
-        {
-            yError() << "[setHessianMatrix] Unable to update the hessian matrix.";
-            return false;
-        }
-    }
-    else
-    {
-        if(!m_optimizerSolver->data()->setHessianMatrix(hessianEigen))
-        {
-            yError() << "[setHessianMatrix] Unable to set first time the hessian matrix.";
-            return false;
-        }
-    }
-    return true;
-}
 
-bool WalkingQPIK_osqp::setGradientVector()
+bool WalkingQPIK_osqp::initializeSolver()
 {
-    iDynTree::Matrix3x3 errorNeckAttitude = iDynTreeHelper::Rotation::skewSymmetric(m_neckOrientation * m_desiredNeckOrientation.inverse());
-
-    // todo
-    if(m_useCoMAsConstraint)
-    {
-        m_gradient = -iDynTree::toEigen(m_neckJacobian).transpose()
-            * iDynTree::toEigen(m_neckWeightMatrix) *
-            m_kAttFoot * (-m_kNeck * iDynTree::unskew(iDynTree::toEigen(errorNeckAttitude)))
-            - iDynTree::toEigen(m_jointRegulatizationGradient) *
-            (iDynTree::toEigen(m_jointRegulatizationGains) * (iDynTree::toEigen(m_regularizationTerm)
-                                                              - iDynTree::toEigen(m_jointPosition)));
-    }
-    else
-    {
-        m_gradient = -iDynTree::toEigen(m_comJacobian).transpose()
-            * iDynTree::toEigen(m_comWeightMatrix) * iDynTree::toEigen(m_comVelocity)
-            - iDynTree::toEigen(m_neckJacobian).transpose() * iDynTree::toEigen(m_neckWeightMatrix) *
-            m_kAttFoot * (-m_kNeck * iDynTree::unskew(iDynTree::toEigen(errorNeckAttitude)))
-            - iDynTree::toEigen(m_jointRegulatizationGradient) *
-            (iDynTree::toEigen(m_jointRegulatizationGains) * (iDynTree::toEigen(m_regularizationTerm)
-                                                              - iDynTree::toEigen(m_jointPosition)));
-    }
-
-    if(m_optimizerSolver->isInitialized())
-    {
-        if(!m_optimizerSolver->updateGradient<Eigen::Dynamic>(m_gradient))
-        {
-            yError() << "[setGradient] Unable to update the gradient.";
-            return false;
-        }
-    }
-    else
+    // Hessian matrix
+    auto hessianDense(iDynTree::toEigen(m_hessianDense));
+    Eigen::SparseMatrix<double> hessianSparse = hessianDense.sparseView();
```

Comment:
To be honest I am not a big fan of ``sparseView`` as it may remove elements which are non-zero but just small. I would prefer to define the sparsity pattern by hand.